data/interface.go

```diff
@@ -20,6 +20,12 @@ const (
 // ModifiedHashes is used to memorize all old hashes and new hashes from when a trie is committed
 type ModifiedHashes map[string]struct{}
 
+// TrieLeaf holds the key to a leaf and the value associated with that leaf
```

Comment:
Exactly at the same pattern I have worked now. We might have a general implementation in core package (keyValueStorage) that will implement a core.KeyValueHolder interface. That interface will be reused in storage package also.

Reply:
Done.